src/Maestro/Maestro.MergePolicies/VersionDetailsPropsMergePolicy.cs

```diff
@@ -0,0 +1,164 @@
+using System.Threading.Tasks;
+using System.Xml;
+using Maestro.MergePolicyEvaluation;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.DarcLib.Helpers;
+
+#nullable enable
+namespace Maestro.MergePolicies;
+public class VersionDetailsPropsMergePolicy : MergePolicy
+{
+    public override string Name => "VersionDetailsProps";
+
+    public override string DisplayName => "Version Details Properties Merge Policy";
+
+    public override async Task<MergePolicyEvaluationResult> EvaluateAsync(PullRequestUpdateSummary pr, IRemote remote)
+    {
+        if (pr.CodeFlowDirection == CodeFlowDirection.ForwardFlow)
+        {
+            // TODO: https://github.com/dotnet/arcade-services/issues/4998 Make the check work for forward flow PRs once we implement the issue
+            return SucceedDecisively("Version Details Properties Merge Policy: Not a backflow PR");
+        }
+
+        XmlDocument versionDetailsProps;
+        try
+        {
+            versionDetailsProps = DependencyFileManager.GetXmlDocument(await remote.GetFileContentsAsync(
+                VersionFiles.VersionDetailsProps,
+                pr.TargetRepoUrl,
+                pr.HeadBranch));
+        }
+        catch (DependencyFileNotFoundException)
+        {
+            // TODO: this should only be in for the transition period until we add VersionDetailsProps to all codeflow repos
+            return SucceedDecisively("Version Details Properties Merge Policy: VersionDetailsProps file not found, skipping validation.");
+        }
+        catch
+        {
+            return FailTransiently("Failed to evaluate VersionDetailsProps merge policy",
+                $"An error occurred while trying to read the {VersionFiles.VersionDetailsProps} file");
+        }
+
+        try
+        {
+            var versionProps = DependencyFileManager.GetXmlDocument(await remote.GetFileContentsAsync(
+                VersionFiles.VersionProps,
+                pr.TargetRepoUrl,
+                pr.HeadBranch));
+
+            // Extract all properties from VersionDetailsProps
+            var versionDetailsPropsProperties = ExtractPropertiesFromXml(versionDetailsProps);
+            
+            // Extract all properties from VersionProps
+            var versionPropsProperties = ExtractPropertiesFromXml(versionProps);
+
+            // Check if any properties from VersionDetailsProps exist in VersionProps
+            var foundProperties = new List<string>();
+            foreach (var versionDetailsPropsProperty in versionDetailsPropsProperties)
+            {
+                if (versionPropsProperties.Contains(versionDetailsPropsProperty))
+                {
+                    foundProperties.Add(versionDetailsPropsProperty);
+                }
+            }
+
+            if (foundProperties.Count > 0)
+            {
+                StringBuilder str = new();
+                str.AppendLine("Properties from `VersionDetailsProps` should not be present in `VersionProps`.");
```

Comment:
I am not sure this is a valid check. I think we were saying we could now start pinning properties by overwriting them in `Versions.props`.

Reply:
why would we need to do that tho? We can just pin them in  Version.Detail.xml or not have them there at all

Reply:
Michael Simons had a point where you don't want to set `Pinned=True` but still want to occasionally receive an update somehow. I don't remember how it went.